lib/lhm/chunker.rb

```diff
@@ -48,8 +47,9 @@ def bottom
       @next_to_insert
     end
 
-    def top(stride)
-      [(@next_to_insert + stride - 1), @limit].min
+    def upper_id(next_id, stride)
+      top = connection.select_value("select id from `#{ origin_name }` where id >= #{ next_id } order by id limit 1 offset #{ stride - 1}")
+      [top ? top.to_i : @limit, @limit].min
```

Comment:
it's confusing to read, at first i thought it wasn't even doing the right thing and you needed ( ) enclosing the ternary, but you don't. 

i'm not coming up with anything much better so... so be it.